adcc/ExcitedStates.py

```diff
@@ -586,3 +654,60 @@ def describe_amplitudes(self, tolerance=0.01, index_format=None):
                 ret += "\n"
                 ret += separator
         return ret[:-1]
+
+    @property
+    def excitations(self):
+        """
+        Provides a list of Excitations, i.e., a view to all individual
+        excited states and their properties. Still under heavy development.
+        """
+        class Excitation:
+            pass
+
+        excitations = []
+        for i in range(self.size):
+            e = Excitation()
+            e.idx = i
+            e.__parent_state = self
+            for key in dir(self):
+                if key == "excitations":
+                    continue
+                if "__" in key or key.startswith("_"):
+                    continue  # skip "private" fields
+                if hasattr(type(self), key):
+                    if not isinstance(getattr(type(self), key), property):
+                        continue
+                else:
+                    continue
+
+                fget = getattr(type(self), key).fget
+                if not hasattr(fget, "__excitation_property"):
+                    # print(dir(fget), key, "not found.")
+                    continue
+                # Extract the kwargs passed to mark_excitation_property
+                kwargs = getattr(fget, "__excitation_property").copy()
+
+                def payload(self, key=key, kwargs=kwargs):
```

Comment:
Payload is not a good name. I just chose it in my trial code.

Reply:
Renamed.